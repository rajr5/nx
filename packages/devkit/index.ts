export { Tree, FileChange } from '@nrwl/tao/src/shared/tree';
export {
  WorkspaceConfiguration,
  TargetConfiguration,
  ProjectConfiguration,
  Generator,
  GeneratorCallback,
  Executor,
  ExecutorContext,
} from '@nrwl/tao/src/shared/workspace';
export {
  NxJsonConfiguration,
  NxJsonProjectConfiguration,
} from '@nrwl/tao/src/shared/nx';
export { logger } from '@nrwl/tao/src/shared/logger';
export { getPackageManagerCommand } from '@nrwl/tao/src/shared/package-manager';
export { runExecutor } from '@nrwl/tao/src/commands/run';

export { formatFiles } from './src/generators/format-files';
export { generateFiles } from './src/generators/generate-files';
export {
  addProjectConfiguration,
  readProjectConfiguration,
  updateProjectConfiguration,
} from './src/generators/project-configuration';
export { toJS } from './src/generators/to-js';

export { readJson, writeJson, updateJson } from './src/utils/json';
export { addDependenciesToPackageJson } from './src/utils/package-json';
export { installPackagesTask } from './src/tasks/install-packages-task';
export { names } from './src/utils/names';
export {
  getWorkspaceLayout,
  getWorkspacePath,
} from './src/utils/get-workspace-layout';
export {
  applyChangesToString,
  ChangeType,
  StringChange,
  StringDeletion,
  StringInsertion,
} from './src/utils/string-change';
export { offsetFromRoot } from './src/utils/offset-from-root';
export { convertNxGenerator } from './src/utils/invoke-nx-generator';
export { convertNxExecutor } from './src/utils/convert-nx-executor';
export { stripIndents } from './src/utils/strip-indents';
