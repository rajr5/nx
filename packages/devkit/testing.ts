export * from './src/create-tree-with-empty-workspace';
