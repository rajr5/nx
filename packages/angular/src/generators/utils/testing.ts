import type { Tree } from '@nrwl/devkit';
import { wrapAngularDevkitSchematic } from '@nrwl/devkit/ngcli-adapter';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

export async function createStorybookTestWorkspaceForLib(
  libName: string
): Promise<Tree> {
  let tree = createTreeWithEmptyWorkspace();

  const libGenerator = wrapAngularDevkitSchematic('@nrwl/angular', 'library');
  const moduleGenerator = wrapAngularDevkitSchematic(
    '@schematics/angular',
    'module'
  );
  const componentGenerator = wrapAngularDevkitSchematic(
    '@schematics/angular',
    'component'
  );

  await libGenerator(tree, { name: libName });

  await componentGenerator(tree, {
    name: 'test-button',
    project: libName,
  });

  tree.write(
    `libs/${libName}/src/lib/test-button/test-button.component.ts`,
    `import { Component, OnInit, Input } from '@angular/core';

export type ButtonStyle = 'default' | 'primary' | 'accent';

@Component({
  selector: 'proj-test-button',
  templateUrl: './test-button.component.html',
  styleUrls: ['./test-button.component.css']
})
export class TestButtonComponent implements OnInit {
  @Input('buttonType') type = 'button';
  @Input() style: ButtonStyle = 'default';
  @Input() age: number;
  @Input() isOn = false;

  constructor() { }

  ngOnInit() {
  }

}`
  );

  tree.write(
    `libs/${libName}/src/lib/test-button/test-button.component.html`,
    `<button [attr.type]="type" [ngClass]="style"></button>`
  );

  const modulePath = `libs/${libName}/src/lib/${libName}.module.ts`;
  tree.write(
    modulePath,
    `import * as ButtonExports from './test-button/test-button.component';
    ${tree.read(modulePath)}`
  );

  // create a module with component that gets exported in a barrel file
  await moduleGenerator(tree, {
    name: 'barrel',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'barrel-button',
    project: libName,
    path: `libs/${libName}/src/lib/barrel`,
    module: 'barrel',
  });

  tree.write(
    `libs/${libName}/src/lib/barrel/barrel-button/index.ts`,
    `export * from './barrel-button.component';`
  );

  tree.write(
    `libs/${libName}/src/lib/barrel/barrel.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BarrelButtonComponent } from './barrel-button';

@NgModule({
  imports: [CommonModule],
  declarations: [BarrelButtonComponent],
})
export class BarrelModule {}`
  );

  // create a module with components that get Angular exported and declared by variable
  await moduleGenerator(tree, {
    name: 'variable-declare',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'variable-declare-button',
    project: libName,
    path: `libs/${libName}/src/lib/variable-declare`,
    module: 'variable-declare',
  });

  await componentGenerator(tree, {
    name: 'variable-declare-view',
    project: libName,
    path: `libs/${libName}/src/lib/variable-declare`,
    module: 'variable-declare',
  });

  tree.write(
    `libs/${libName}/src/lib/variable-declare/variable-declare.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VariableDeclareButtonComponent } from './variable-declare-button/variable-declare-button.component';
import { VariableDeclareViewComponent } from './variable-declare-view/variable-declare-view.component';

const COMPONENTS = [
  VariableDeclareButtonComponent,
  VariableDeclareViewComponent
]

@NgModule({
  imports: [CommonModule],
  declarations: COMPONENTS,
  exports: COMPONENTS
})
export class VariableDeclareModule {}`
  );

  // create another button in a nested subpath
  await moduleGenerator(tree, {
    name: 'nested',
    project: libName,
    path: `libs/${libName}/src/lib`,
  });

  await componentGenerator(tree, {
    name: 'nested-button',
    project: libName,
    module: 'nested',
    path: `libs/${libName}/src/lib/nested`,
  });

  await componentGenerator(tree, {
    name: 'test-other',
    project: libName,
  });

  return tree;
}
