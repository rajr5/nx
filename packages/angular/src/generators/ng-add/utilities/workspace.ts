import {
  generateFiles,
  joinPathFragments,
  NxJsonConfiguration,
  readJson,
  readWorkspaceConfiguration,
  Tree,
  updateJson,
  updateWorkspaceConfiguration,
  writeJson,
} from '@nrwl/devkit';
import { DEFAULT_NRWL_PRETTIER_CONFIG } from '@nrwl/workspace/src/generators/workspace/workspace';
import { deduceDefaultBase } from '@nrwl/workspace/src/utilities/default-base';
import { resolveUserExistingPrettierConfig } from '@nrwl/workspace/src/utilities/prettier';
import { getRootTsConfigPathInTree } from '@nrwl/workspace/src/utilities/typescript';
import { prettierVersion } from '@nrwl/workspace/src/utils/versions';
import { readFileSync } from 'fs';
import { dirname, join } from 'path';
import { angularDevkitVersion, nxVersion } from '../../../utils/versions';
import { GeneratorOptions } from '../schema';
import {
  getCypressConfigFile,
  getE2eKey,
  getE2eProject,
  isCypressE2eProject,
  isProtractorE2eProject,
} from './e2e-utils';
import { WorkspaceProjects } from './types';

// TODO: most of the validation here will be moved to the app migrator when
// support for multiple apps is added. This will only contain workspace-wide
// validation.
export function validateWorkspace(
  tree: Tree,
  projects: WorkspaceProjects
): void {
  try {
    if (!tree.exists('package.json')) {
      throw new Error('Cannot find package.json');
    }

    if (!tree.exists('angular.json')) {
      throw new Error('Cannot find angular.json');
    }

    const e2eKey = getE2eKey(projects);
    const e2eApp = getE2eProject(projects);

    if (!e2eApp) {
      return;
    }

    if (isProtractorE2eProject(e2eApp.config)) {
      if (tree.exists(e2eApp.config.targets.e2e.options.protractorConfig)) {
        return;
      }

      console.info(
        `Make sure the "${e2eKey}.architect.e2e.options.protractorConfig" is valid or the "${e2eKey}" project is removed from "angular.json".`
      );
      throw new Error(
        `An e2e project with Protractor was found but "${e2eApp.config.targets.e2e.options.protractorConfig}" could not be found.`
      );
    }

    if (isCypressE2eProject(e2eApp.config)) {
      const configFile = getCypressConfigFile(e2eApp.config);
      if (configFile && !tree.exists(configFile)) {
        throw new Error(
          `An e2e project with Cypress was found but "${configFile}" could not be found.`
        );
      }

      if (!tree.exists('cypress')) {
        throw new Error(
          `An e2e project with Cypress was found but the "cypress" directory could not be found.`
        );
      }

      return;
    }

    throw new Error(
      `An e2e project was found but it's using an unsupported executor "${e2eApp.config.targets.e2e.executor}".`
    );
  } catch (e) {
    console.error(e.message);
    console.error(
      'Your workspace could not be converted into an Nx Workspace because of the above error.'
    );
    throw e;
  }
}

export function createNxJson(
  tree: Tree,
  options: GeneratorOptions,
  setWorkspaceLayoutAsNewProjectRoot: boolean = false
): void {
  const { projects = {}, newProjectRoot = '' } = readJson(tree, 'angular.json');
  // TODO: temporarily leaving this here because it's the old behavior for a
  // minimal migration, will be removed in a later PR
  const hasLibraries = Object.keys(projects).find(
    (project) =>
      projects[project].projectType &&
      projects[project].projectType !== 'application'
  );
  if (Object.keys(projects).length !== 1 || hasLibraries) {
    throw new Error(
      `The schematic can only be used with Angular CLI workspaces with a single application.`
    );
  }

  writeJson<NxJsonConfiguration>(tree, 'nx.json', {
    npmScope: options.npmScope,
    affected: {
      defaultBase: options.defaultBase ?? deduceDefaultBase(),
    },
    implicitDependencies: {
      'package.json': {
        dependencies: '*',
        devDependencies: '*',
      },
      '.eslintrc.json': '*',
    },
    tasksRunnerOptions: {
      default: {
        runner: 'nx/tasks-runners/default',
        options: {
          cacheableOperations: ['build', 'lint', 'test', 'e2e'],
        },
      },
    },
    targetDependencies: {
      build: [
        {
          target: 'build',
          projects: 'dependencies',
        },
      ],
    },
    workspaceLayout: setWorkspaceLayoutAsNewProjectRoot
      ? { appsDir: newProjectRoot, libsDir: newProjectRoot }
      : undefined,
  });
}

export function decorateAngularCli(tree: Tree): void {
  const nrwlWorkspacePath = require.resolve('@nrwl/workspace/package.json');
  const decorateCli = readFileSync(
    join(
      dirname(nrwlWorkspacePath),
      'src/generators/utils/decorate-angular-cli.js__tmpl__'
    ),
    'utf-8'
  );
  tree.write('decorate-angular-cli.js', decorateCli);

  updateJson(tree, 'package.json', (json) => {
    if (
      json.scripts &&
      json.scripts.postinstall &&
      !json.scripts.postinstall.includes('decorate-angular-cli.js')
    ) {
      // if exists, add execution of this script
      json.scripts.postinstall += ' && node ./decorate-angular-cli.js';
    } else {
      if (!json.scripts) json.scripts = {};
      // if doesn't exist, set to execute this script
      json.scripts.postinstall = 'node ./decorate-angular-cli.js';
    }
    if (json.scripts.ng) {
      json.scripts.ng = 'nx';
    }
    return json;
  });
}

export function updateWorkspaceConfigDefaults(tree: Tree): void {
  const workspaceConfig = readWorkspaceConfiguration(tree);
  delete (workspaceConfig as any).newProjectRoot;
  workspaceConfig.cli = workspaceConfig.cli ?? {};
  if (!workspaceConfig.cli.defaultCollection) {
    workspaceConfig.cli.defaultCollection = '@nrwl/angular';
  }
  updateWorkspaceConfiguration(tree, workspaceConfig);
}

export function updateRootTsConfig(tree: Tree): void {
  const tsconfig = readJson(tree, getRootTsConfigPathInTree(tree));
  tsconfig.compilerOptions.paths ??= {};
  tsconfig.compilerOptions.baseUrl = '.';
  tsconfig.compilerOptions.rootDir = '.';
  tsconfig.exclude = Array.from(
    new Set([...(tsconfig.exclude ?? []), 'node_modules', 'tmp'])
  );
  writeJson(tree, 'tsconfig.base.json', tsconfig);

  if (tree.exists('tsconfig.json')) {
    tree.delete('tsconfig.json');
  }
}

export function updatePackageJson(tree: Tree): void {
  updateJson(tree, 'package.json', (packageJson) => {
    packageJson.scripts = packageJson.scripts ?? {};
    Object.keys(packageJson.scripts).forEach((script) => {
      packageJson.scripts[script] = packageJson.scripts[script]
        .replace(/^ng /, 'nx ')
        .replace(/ ng /, ' nx ');
    });

    packageJson.devDependencies = packageJson.devDependencies ?? {};
    packageJson.dependencies = packageJson.dependencies ?? {};
    if (!packageJson.devDependencies['@angular/cli']) {
      packageJson.devDependencies['@angular/cli'] = angularDevkitVersion;
    }
    if (!packageJson.devDependencies['@nrwl/workspace']) {
      packageJson.devDependencies['@nrwl/workspace'] = nxVersion;
    }
    if (!packageJson.devDependencies['nx']) {
      packageJson.devDependencies['nx'] = nxVersion;
    }
    if (!packageJson.devDependencies['prettier']) {
      packageJson.devDependencies['prettier'] = prettierVersion;
    }

    return packageJson;
  });
}

export function updateTsLint(tree: Tree): void {
  if (!tree.exists(`tslint.json`)) {
    return;
  }

  updateJson(tree, 'tslint.json', (tslintJson) => {
    [
      'no-trailing-whitespace',
      'one-line',
      'quotemark',
      'typedef-whitespace',
      'whitespace',
    ].forEach((key) => {
      tslintJson[key] = undefined;
    });
    tslintJson.rulesDirectory = tslintJson.rulesDirectory ?? [];
    tslintJson.rulesDirectory.push('node_modules/@nrwl/workspace/src/tslint');
    tslintJson.rules['nx-enforce-module-boundaries'] = [
      true,
      {
        allow: [],
        depConstraints: [{ sourceTag: '*', onlyDependOnLibsWithTags: ['*'] }],
      },
    ];
    return tslintJson;
  });
}

export async function createWorkspaceFiles(tree: Tree): Promise<void> {
  updateVsCodeRecommendedExtensions(tree);
  await updatePrettierConfig(tree);

  generateFiles(tree, joinPathFragments(__dirname, '../files/root'), '.', {
    tmpl: '',
    dot: '.',
    rootTsConfigPath: getRootTsConfigPathInTree(tree),
  });
}

export function createRootKarmaConfig(tree: Tree): void {
  generateFiles(
    tree,
    joinPathFragments(__dirname, '../files/root-karma'),
    '.',
    {
      tmpl: '',
    }
  );
}

function updateVsCodeRecommendedExtensions(tree: Tree): void {
  const recommendations = [
    'nrwl.angular-console',
    'angular.ng-template',
    'dbaeumer.vscode-eslint',
    'esbenp.prettier-vscode',
  ];
  if (tree.exists('.vscode/extensions.json')) {
    updateJson(
      tree,
      '.vscode/extensions.json',
      (json: { recommendations?: string[] }) => {
        json.recommendations = json.recommendations || [];
        recommendations.forEach((extension) => {
          if (!json.recommendations.includes(extension)) {
            json.recommendations.push(extension);
          }
        });

        return json;
      }
    );
  } else {
    writeJson(tree, '.vscode/extensions.json', {
      recommendations,
    });
  }
}

async function updatePrettierConfig(tree: Tree): Promise<void> {
  const existingPrettierConfig = await resolveUserExistingPrettierConfig();
  if (!existingPrettierConfig) {
    writeJson(tree, '.prettierrc', DEFAULT_NRWL_PRETTIER_CONFIG);
  }

  if (!tree.exists('.prettierignore')) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, '../files/prettier'),
      '.',
      { tmpl: '', dot: '.' }
    );
  }
}
