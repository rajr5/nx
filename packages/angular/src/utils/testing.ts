import { join } from 'path';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { Rule, Tree } from '@angular-devkit/schematics';

const testRunner = new SchematicTestRunner(
  '@nrwl/angular',
  join(__dirname, '../../generators.json')
);

testRunner.registerCollection(
  '@nrwl/jest',
  join(__dirname, '../../../jest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/workspace',
  join(__dirname, '../../../workspace/generators.json')
);

testRunner.registerCollection(
  '@nrwl/cypress',
  join(__dirname, '../../../cypress/generators.json')
);

testRunner.registerCollection(
  '@nrwl/storybook',
  join(__dirname, '../../../storybook/generators.json')
);

const migrationTestRunner = new SchematicTestRunner(
  '@nrwl/workspace',
  join(__dirname, '../../migrations.json')
);

/**
 * @deprecated will be removed in v13.
 */
export function runSchematic<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}

export function runMigration<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return migrationTestRunner
    .runSchematicAsync(schematicName, options, tree)
    .toPromise();
}

/**
 * @deprecated will be removed in v13.
 */
export function runExternalSchematic<SchemaOptions = any>(
  collectionName: string,
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return testRunner
    .runExternalSchematicAsync(collectionName, schematicName, options, tree)
    .toPromise();
}

export function callRule(rule: Rule, tree: Tree) {
  return testRunner.callRule(rule, tree).toPromise();
}
