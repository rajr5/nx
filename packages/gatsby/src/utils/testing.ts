import { Tree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { join } from 'path';

const testRunner = new SchematicTestRunner(
  '@nrwl/gatsby',
  join(__dirname, '../../collection.json')
);

testRunner.registerCollection(
  '@nrwl/jest',
  join(__dirname, '../../../jest/collection.json')
);

testRunner.registerCollection(
  '@nrwl/cypress',
  join(__dirname, '../../../cypress/collection.json')
);

testRunner.registerCollection(
  '@nrwl/react',
  join(__dirname, '../../../react/collection.json')
);

export function runSchematic(schematicName: string, options: any, tree: Tree) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}
