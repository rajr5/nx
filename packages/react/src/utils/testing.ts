import { join } from 'path';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { Tree } from '@angular-devkit/schematics';

const testRunner = new SchematicTestRunner(
  '@nrwl/react',
  join(__dirname, '../../collection.json')
);

export function runSchematic(schematicName: string, options: any, tree: Tree) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}
