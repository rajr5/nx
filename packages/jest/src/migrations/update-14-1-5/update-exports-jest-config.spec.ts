import {
  readProjectConfiguration,
  stripIndents,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { createTree, createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { libraryGenerator as workspaceLib } from '@nrwl/workspace';
import {
  updateExportsJestConfig,
  updateRootFiles,
  updateToDefaultExport,
} from './update-exports-jest-config';

describe('Jest Migration (v14.1.2)', () => {
  let tree: Tree;
  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace(2);
  });

  it('should update root jest files', () => {
    tree.write(
      'jest.config.ts',
      stripIndents`
      const { getJestProjects } = require('@nrwl/jest');

      module.exports = {
        projects: getJestProjects()
      };`
    );

    tree.write(
      'jest.preset.ts',
      stripIndents`
      const nxPreset = require('@nrwl/jest/preset');

      module.exports = { ...nxPreset };`
    );

    const status = updateRootFiles(tree);

    expect(status).toEqual({ didUpdateRootPreset: true });
    expect(tree.read('jest.config.ts', 'utf-8')).toEqual(stripIndents`
      const { getJestProjects } = require('@nrwl/jest');

      export default {
        projects: getJestProjects()
      };
    `);
    expect(tree.read('jest.preset.js', 'utf-8')).toEqual(stripIndents`
      const nxPreset = require('@nrwl/jest/preset').default;

      module.exports = { ...nxPreset };`);
  });

  it('should update individual project jest configs', async () => {
    await workspaceLib(tree, { name: 'lib-one' });
    tree.rename('jest.preset.js', 'jest.preset.ts');
    tree.write(
      'libs/lib-one/jest.config.ts',
      `
const nxPreset = require('@nrwl/jest/preset');
const someOtherImport = require('../something/else.js');
module.exports = {
  ...someOtherImport,
  ...nxPreset,
  displayName: 'lib-one',
  preset: '../../jest.preset.ts',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
      stringifyContentPathRegex: '\\\\.(html|svg)$',
    },
  },
  coverageDirectory: '../../coverage/apps/lib-one',
  transform: {
    '^.+\\\\.(ts|mjs|js|html)$': 'jest-preset-angular',
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
};
`
    );
    updateExportsJestConfig(tree);

    const config = tree.read('libs/lib-one/jest.config.ts', 'utf-8');
    expect(config).toMatchSnapshot();
  });

  it('should work with multiple configurations', async () => {
    await workspaceLib(tree, { name: 'lib-one' });
    tree.rename('jest.preset.js', 'jest.preset.ts');
    updateProjectConfiguration(tree, 'lib-one', {
      ...readProjectConfiguration(tree, 'lib-one'),
      targets: {
        test: {
          executor: '@nrwl/jest:jest',
          options: {
            jestConfig: 'libs/lib-one/jest.config.ts',
            passWithoutTests: true,
          },
          configurations: {
            production: {
              silent: true,
            },
          },
        },
      },
    });

    tree.write(
      'libs/lib-one/jest.config.ts',
      `
const nxPreset = require('@nrwl/jest/preset');
const someOtherImport = require('../something/else.js');
module.exports = {
  ...someOtherImport,
  ...nxPreset,
  displayName: 'lib-one',
  preset: '../../jest.preset.ts',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
      stringifyContentPathRegex: '\\\\.(html|svg)$',
    },
  },
  coverageDirectory: '../../coverage/apps/lib-one',
  transform: {
    '^.+\\\\.(ts|mjs|js|html)$': 'jest-preset-angular',
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
};
`
    );

    updateExportsJestConfig(tree);

    const config = tree.read('libs/lib-one/jest.config.ts', 'utf-8');
    expect(config).toMatchSnapshot();
    expect(tree.exists('jest.preset.ts')).toBeFalsy();
    expect(tree.exists('jest.preset.js')).toBeTruthy();
  });

  it('should convert module.exports => export default', () => {
    tree = createTree();

    tree.write(
      'jest.config.js',
      stripIndents`
  const { getJestProjects } = require('@nrwl/jest');
  const nxPreset = require('@nrwl/jest/preset');
  
  
  const someFn = () => ({more: 'stuff'});
  module.export.abc = someFn;
  module.exports = {
    ...nxPreset,
    more: 'stuff',
    someFn,
    projects: getJestProjects()
  };`
    );
    updateToDefaultExport(tree, 'jest.config.js');

    expect(tree.read('jest.config.js', 'utf-8')).toMatchSnapshot();
  });
});
