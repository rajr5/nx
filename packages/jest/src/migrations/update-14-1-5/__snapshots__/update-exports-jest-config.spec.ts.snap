// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Migration (v14.1.2) should convert module.exports => export default 1`] = `
"const { getJestProjects } = require('@nrwl/jest');
const nxPreset = require('@nrwl/jest/preset');


const someFn = () => ({more: 'stuff'});
module.export.abc = someFn;
export default {
...nxPreset,
more: 'stuff',
someFn,
projects: getJestProjects()
};"
`;

exports[`Jest Migration (v14.1.2) should update individual project jest configs 1`] = `
"
const nxPreset = require('@nrwl/jest/preset').default;
const someOtherImport = require('../something/else.js');
export default {
  ...someOtherImport,
  ...nxPreset,
  displayName: 'lib-one',
  preset: '../../jest.preset.js',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
      stringifyContentPathRegex: '\\\\\\\\.(html|svg)$',
    },
  },
  coverageDirectory: '../../coverage/apps/lib-one',
  transform: {
    '^.+\\\\\\\\.(ts|mjs|js|html)$': 'jest-preset-angular',
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\\\\\\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
};
"
`;

exports[`Jest Migration (v14.1.2) should work with multiple configurations 1`] = `
"
const nxPreset = require('@nrwl/jest/preset').default;
const someOtherImport = require('../something/else.js');
export default {
  ...someOtherImport,
  ...nxPreset,
  displayName: 'lib-one',
  preset: '../../jest.preset.js',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
      stringifyContentPathRegex: '\\\\\\\\.(html|svg)$',
    },
  },
  coverageDirectory: '../../coverage/apps/lib-one',
  transform: {
    '^.+\\\\\\\\.(ts|mjs|js|html)$': 'jest-preset-angular',
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\\\\\\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
};
"
`;
